### PR TITLE
[Shipping Labels Revamp] Handle Package selection on the main form

### LINKED ISSUE #12747: [Shipping Labels: Packages] Add Package Opening UI Section
Summary
==========
Once the Package UI flow is functional, we should allow it to be actioned from the main Shipping Labels creation flow, this task focus on this feature by introducing an opening view within the main form.

Designs
==========
Image
Image

### PR DESCRIPTION
**Pull request by ThomazFB**
Summary
==========
Fix issue #12747 by introducing the complete handling of the Package selection into the main Shipping Labels form.

Screen Capture
==========
https://github.com/user-attachments/assets/bee15145-bdcf-4102-88c4-109f1951e596

How to Test
==========

## Scenario 1 - Carrier Package

1. Go to the Order Details of an Order eligible for Shipping Label Creation
2. Hit the Create Shipping Labels button
3. Once inside the new Shipping Labels form, hit the `Select a Package` button
4. Select a Carrier Package and hit the `Add Package` Button
5. Verify that the Shipping Label form is presented back, with the Carrier package displayed properly
6. Verify that the subtitle of the Package contains the Carrier Category of that given package
7. Verify that the edit button re-enters the Package selection UI

## Scenario 2 - Saved Package

1. Go to the Order Details of an Order eligible for Shipping Label Creation
2. Hit the Create Shipping Labels button
3. Once inside the new Shipping Labels form, hit the `Select a Package` button
4. Select a Saved Package and hit the `Add Package` Button
5. Verify that the Shipping Label form is presented back, with the Saved package displayed properly
6. Verify that the subtitle of the Package contains the Package type (Box or Envelope)
7. Verify that the edit button re-enters the Package selection UI

## Scenario 3 - Custom Package not saved as a template

1. Go to the Order Details of an Order eligible for Shipping Label Creation
2. Hit the Create Shipping Labels button
3. Once inside the new Shipping Labels form, hit the `Select a Package` button
4. Fill the Custom Package form, DO NOT select the `save as template` check and hit the `Add Package` Button
5. Verify that the Shipping Label form is presented back, with the Custom package displayed properly
6. Verify that the Package subtitle contains the Package type (Box or Envelope)
7. Verify that the Package title is named as `Custom Package` 
8. Verify that the edit button re-enters the Package selection UI

## Scenario 4 - Custom Package saved as a template

1. Go to the Order Details of an Order eligible for Shipping Label Creation
2. Hit the Create Shipping Labels button
3. Once inside the new Shipping Labels form, hit the `Select a Package` button
4. Fill the Custom Package form, SELECT the `save as template` check
5. Add the package weight and name and hit the `Add Package` Button
6. Verify that the Shipping Label form is presented back, with the Custom package displayed properly
7. Verify that the Package subtitle contains the Package type (Box or Envelope)
8. Verify that the Package title is named as the given name  in the Custom package form
9. Verify that the edit button re-enters the Package selection UI

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/WooShippingLabelCreationScreen.kt [hunk 1]
```diff
@@ -395,14 +389,86 @@ private fun PackageCard(
...
+            Text(
+                text = stringResource(id = R.string.shipping_label_package_selected_title),
+                style = MaterialTheme.typography.subtitle1,
+                fontWeight = FontWeight.SemiBold
+            )
+            IconButton(
+                onClick = onSelectPackageClick
+            ) {
+                Icon(
+                    painter = painterResource(id = R.drawable.ic_edit),
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/WooShippingLabelCreationScreen.kt]: np: We should use the material pencil here

**Review by atorresveiga — APPROVED** [review]
LGTM! :shipit: 

Nice work @ThomazFB! Everything is working as described. I left a small observation about the pencil icon, but is not a blocker for this PR

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/WooShippingLabelCreationScreen.kt (hunk 1 above)
Comment (ThomazFB) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/WooShippingLabelCreationScreen.kt]: Oh, that's true! Thanks for the suggestion, it's fixed in 5aff9b32aff8c802950064003fff935aad6582f8

**Pull request merged**